ckan/templates/snippets/activity_stream.html

```diff
@@ -0,0 +1,51 @@
+{% macro actor(activity) %}
+<span class="actor">
+  {{ h.linked_user(activity.user_id, 0, 30) }}
+</span>
+{% endmacro %}
+
+{% macro dataset(activity) %}
+<span class="dataset">
+  {{ h.dataset_link(activity.data.package) }}
+</span>
+{% endmacro %}
+
+{% macro organization(activity) %}
+{{ h.organization_link(activity.data.group) }}
+{% endmacro %}
+
+{% macro user(activity) %}
+<span class="actor">
+  {{ h.linked_user(activity.object_id, 0, 20) }}
+</span>
+{% endmacro %}
+
+{% macro tag(activity) %}
+<span class="actor">
+  {{ h.tag_link(activity.data.tag) }}
+</span>
+{% endmacro %}
+
+{% macro group(activity) %}
+<span class="group">
+  {{ h.group_link(activity.data.group) }}
+</span>
+{% endmacro %}
```

Comment:
This looks like these macros make it easy and efficient to override the style of the links created, just by extending this template. Very nice

Reply:
Yup, record activities of any type you want and just add a template snippet to render them in your extension (or in core). The goal for activities is to be as simple as possible to use.